### ISSUE TITLE
Error: Cannot compute gradient: gradient function not found for onesLike.

### ISSUE THREAD

**Issue by vanbasten17** [tensorflow/tfjs#1398]
#### TensorFlow.js version
@tensorflow/tfjs: 1.0.1
@tensorflow/tfjs-node: 1.0.1

#### Browser version
NodeJS  (v8.15.1)

#### Describe the problem or feature request
After upgrading from tfjs (v 0.15.3) and tfjs-node (v 0.3.2) to the latest release, I am not able to fit the previously working model anymore. 

UnhandledPromiseRejectionWarning: Error: Cannot compute gradient: gradient function not found for onesLike.
    at _loop_1 (/tfjs1.0/node_modules/@tensorflow/tfjs-core/dist/tape.js:128:19)
    at Object.backpropagateGradients (/tfjs1.0/node_modules/@tensorflow/tfjs-core/dist/tape.js:165:9)
    at /tfjs1.0/node_modules/@tensorflow/tfjs-core/dist/engine.js:420:20
    at /tfjs1.0/node_modules/@tensorflow/tfjs-core/dist/engine.js:114:22
    at Engine.scopedRun (/tfjs1.0/node_modules/@tensorflow/tfjs-core/dist/engine.js:124:23)
    at Engine.tidy (/tfjs1.0/node_modules/@tensorflow/tfjs-core/dist/engine.js:113:21)
    at Engine.gradients (/tfjs1.0/node_modules/@tensorflow/tfjs-core/dist/engine.js:407:21)
    at Object.variableGrads (/tfjs1.0/node_modules/@tensorflow/tfjs-core/dist/gradients.js:258:39)
    at AdamOptimizer.Optimizer.computeGradients (/tfjs1.0/node_modules/@tensorflow/tfjs-core/dist/optimizers/optimizer.js:80:26)
    at AdamOptimizer.Optimizer.minimize (/tfjs1.0/node_modules/@tensorflow/tfjs-core/dist/optimizers/optimizer.js:55:23)

After trying to debug the code (and as far as I can understand) it seems that there's a mismatching node in the different versions:

**Previous Version:**
{ id: 68,
  name: 'mul',
  inputs:
   { '$a':
      Tensor {
        isDisposedInternal: false,
        shape: [Array],
        dtype: 'float32',
        size: 1600,
        strides: [Array],
        dataId: {},
        id: 26648,
        rankType: '2' } },
  outputs:
   [ Tensor {
       isDisposedInternal: false,
       shape: [Array],
       dtype: 'float32',
       size: 1600,
       strides: [Array],
       dataId: {},
       id: 26819,
       rankType: '2' } ],
  gradient: [Function] }

**Current Version:**
{ id: 12,
  name: 'onesLike',
  inputs:
   { '$x':
      Tensor {
        isDisposedInternal: false,
        shape: [Array],
        dtype: 'float32',
        size: 1600,
        strides: [Array],
        dataId: {},
        id: 26648,
        rankType: '2' } },
  outputs:
   [ Tensor {
       isDisposedInternal: false,
       shape: [Array],
       dtype: 'float32',
       size: 1600,
       strides: [Array],
       dataId: {},
       id: 26698,
       rankType: '2' } ] }

#### Code to reproduce the bug / link to feature request
  const model = tf.sequential()
  model.add(
    tf.layers.embedding({
      inputDim: Object.keys(word_index).length, //616
      outputDim: EMBEDDING_DIM, //50
      inputLength: MAX_SEQUENCE_LENGTH //50
    })
  )
  model.layers[0].weights = embedding_matrix
  model.layers[0].trainable = true
  model.add(
    tf.layers.lstm({
      units: UNITS, //100
      dropout: DROPOUT_REG, // 0.2
      recurrentDropout: DROPOUT_REG //0.2
    })
  )
  model.add(
    tf.layers.dense({
      units: Object.keys(intents).length, //9
      activation: 'softmax'
    })
  )
  model.summary()

  model.compile({
    optimizer: tf.train.adam(LEARNING_RATE), 
    loss: 'categoricalCrossentropy',
    metrics: ['accuracy']
  })
  const history = await model.fit(tensor_data, tensor_labels, {
    epochs: EPOCHS,
    validationSplit: VAL_SPLIT
  })

**Comment by BenjaminWegener** [comment]
Its because onesLike and zerosLike were converted to KERNEL version. replace the functions with their deprecated versions while we wait for an update.

`/*			function zerosLike_(e) {
				var t = convertToTensor(e, "x", "zerosLike");
				return ENV.engine.runKernel(function (e) {
					return e.zerosLike(t);
				}, {
					$x: t
				}, null);
			} */
			function zerosLike_(e) {
				var t = convertToTensor(e, "x", "zerosLike");
				return zeros(t.shape, t.dtype);
			}`

**Comment by d3x0r** [comment]
How do you rebuild tfjs-core on windows after making that change?   or make similar change to compiled .esm.js?
```
,Xn=An({zerosLike_:function(t){var e=wn(t,"x","zerosLike");return Dt.runKernel(function(t){return t.zerosLike(e)},{$x:e},null)}})
```

**Comment by BenjaminWegener** [comment]
just sorta hacked the tf.js browser version, for node.js didn't try
